### PR TITLE
fix: #2206 indexing regexp

### LINKED ISSUE #2206: Indexing a RegExp crashes the compiler
## Code
```ts
/ /[0]
```
[Try it in the playground](https://www.assemblyscript.org/editor.html#IyFvcHRpbWl6ZT1zcGVlZCZydW50aW1lPXN0dWIKLyAvWzBdCiMhaHRtbAo8dGV4dGFyZWEgaWQ9Im91dHB1dCIgc3R5bGU9ImhlaWdodDogMTAwJTsgd2lkdGg6IDEwMCUiIHJlYWRvbmx5PjwvdGV4dGFyZWE+CjxzY3JpcHQ+CmxvYWRlci5pbnN0YW50aWF0ZShtb2R1bGVfd2FzbSwgeyAvKiBpbXBvcnRzICovIH0pCiAgLnRoZW4oKHsgZXhwb3J0cyB9KSA9PiB7CiAgICBjb25zdCBvdXRwdXQgPSBkb2N1bWVudC5nZXRFbGVtZW50QnlJZCgnb3V0cHV0JykKICAgIGZvciAobGV0IGkgPSAwOyBpIDw9IDEwOyArK2kpIHsKICAgICAgb3V0cHV0LnZhbHVlICs9IGBmaWIoJHtpfSkgPSAke2V4cG9ydHMuZmliKGkpfVxuYAogICAgfQogIH0pCjwvc2NyaXB0Pg==)
## Stack trace
```
▌ Whoops, the AssemblyScript compiler has crashed during compile :-(
▌ 
▌ Here is the stack trace hinting at the problem, perhaps it's useful?
▌ 
▌ AssertionError: assertion failed
▌     at assert (D:\assemblyscript\std\portable\index.js:200:11)
▌     at Resolver.lookupLiteralExpression (D:\assemblyscript\src\resolver.ts:2253:5)
▌     at Resolver.resolveLiteralExpression (D:\assemblyscript\src\resolver.ts:2268:24)
▌     at Resolver.resolveExpression (D:\assemblyscript\src\resolver.ts:1078:21)
▌     at Compiler.compileElementAccessExpression (D:\assemblyscript\src\compiler.ts:7412:36)
▌     at Compiler.compileExpression (D:\assemblyscript\src\compiler.ts:3498:21)
▌     at Compiler.compileExpressionStatement (D:\assemblyscript\src\compiler.ts:2521:17)
▌     at Compiler.compileStatement (D:\assemblyscript\src\compiler.ts:2213:21)
▌     at Compiler.compileTopLevelStatement (D:\assemblyscript\src\compiler.ts:2175:25)
▌     at Compiler.compileFile (D:\assemblyscript\src\compiler.ts:1111:12)
▌     at Compiler.compile (D:\assemblyscript\src\compiler.ts:486:14)
▌     at Object.compile (D:\assemblyscript\src\index.ts:280:32)
▌     at D:\assemblyscript\cli\asc.js:902:31
▌     at measure (D:\assemblyscript\cli\asc.js:1409:3)
▌     at Object.main (D:\assemblyscript\cli\asc.js:900:24)
▌ 
▌ If it refers to the dist files, try to 'npm install source-map-support' and
▌ run again, which should then show the actual code location in the sources.
▌ 
▌ If you see where the error is, feel free to send us a pull request. If not,
▌ please let us know: https://github.com/AssemblyScript/assemblyscript/issues
▌ 
▌ Thank you!
```

### LINKED ISSUE #2207: `typeof` a RegExp crashes the compiler
## Code
```ts
typeof / /
```
[Try it in the playground](https://www.assemblyscript.org/editor.html#IyFvcHRpbWl6ZT1zcGVlZCZydW50aW1lPXN0dWIKdHlwZW9mIC8gLwojIWh0bWwKPHRleHRhcmVhIGlkPSJvdXRwdXQiIHN0eWxlPSJoZWlnaHQ6IDEwMCU7IHdpZHRoOiAxMDAlIiByZWFkb25seT48L3RleHRhcmVhPgo8c2NyaXB0Pgpsb2FkZXIuaW5zdGFudGlhdGUobW9kdWxlX3dhc20sIHsgLyogaW1wb3J0cyAqLyB9KQogIC50aGVuKCh7IGV4cG9ydHMgfSkgPT4gewogICAgY29uc3Qgb3V0cHV0ID0gZG9jdW1lbnQuZ2V0RWxlbWVudEJ5SWQoJ291dHB1dCcpCiAgICBmb3IgKGxldCBpID0gMDsgaSA8PSAxMDsgKytpKSB7CiAgICAgIG91dHB1dC52YWx1ZSArPSBgZmliKCR7aX0pID0gJHtleHBvcnRzLmZpYihpKX1cbmAKICAgIH0KICB9KQo8L3NjcmlwdD4=)
## Stack trace
```
▌ Whoops, the AssemblyScript compiler has crashed during compile :-(
▌ 
▌ Here is the stack trace hinting at the problem, perhaps it's useful?
▌ 
▌ AssertionError: assertion failed
▌     at assert (D:\assemblyscript\std\portable\index.js:200:11)
▌     at Resolver.lookupLiteralExpression (D:\assemblyscript\src\resolver.ts:2253:5)
▌     at Resolver.lookupExpression (D:\assemblyscript\src\resolver.ts:959:21)
▌     at Compiler.compileTypeof (D:\assemblyscript\src\compiler.ts:9930:35)
▌     at Compiler.compileUnaryPrefixExpression (D:\assemblyscript\src\compiler.ts:9889:21)
▌     at Compiler.compileExpression (D:\assemblyscript\src\compiler.ts:3539:21)
▌     at Compiler.compileExpressionStatement (D:\assemblyscript\src\compiler.ts:2521:17)
▌     at Compiler.compileStatement (D:\assemblyscript\src\compiler.ts:2213:21)
▌     at Compiler.compileTopLevelStatement (D:\assemblyscript\src\compiler.ts:2175:25)
▌     at Compiler.compileFile (D:\assemblyscript\src\compiler.ts:1111:12)
▌     at Compiler.compile (D:\assemblyscript\src\compiler.ts:486:14)
▌     at Object.compile (D:\assemblyscript\src\index.ts:280:32)
▌     at D:\assemblyscript\cli\asc.js:902:31
▌     at measure (D:\assemblyscript\cli\asc.js:1409:3)
▌     at Object.main (D:\assemblyscript\cli\asc.js:900:24)
▌ 
▌ If it refers to the dist files, try to 'npm install source-map-support' and
▌ run again, which should then show the actual code location in the sources.
▌ 
▌ If you see where the error is, feel free to send us a pull request. If not,
▌ please let us know: https://github.com/AssemblyScript/assemblyscript/issues
▌ 
▌ Thank you!
```

### PR DESCRIPTION
**Pull request by technohippy**
⯈fix #2206 
⯈fix #2207

- [x] I've read the contributing guidelines
- [x] I've added my name and email to the NOTICE file

### CODE REVIEW

File: src/resolver.ts [hunk 1]
```diff
@@ -2249,6 +2249,14 @@ export class Resolver extends DiagnosticEmitter {
         }
         return null;
       }
+      case LiteralKind.REGEXP: {
+        this.error(
```
Comment (romdotdog) [inline src/resolver.ts:2253]: Missing `if (reportMode == ReportMode.REPORT) {` here?

Comment (technohippy) [reply src/resolver.ts]: Thanks, I've fixed.

File: src/resolver.ts [hunk 2]
```diff
@@ -2249,6 +2249,17 @@ export class Resolver extends DiagnosticEmitter {
...
+      case LiteralKind.REGEXP: {
+        if (reportMode == ReportMode.REPORT) {
+          this.error(
+            DiagnosticCode.Not_implemented_0,
+            node.range,
+            "Regular expressions"
+          );
+        }
+        return null;
+        return null;
```
Comment (dcodeIO) [inline src/resolver.ts:2261]:
```suggestion
```

**Comment by dcodeIO** [comment]
While this introduces an error, it seems that what's missing here is resolving regexp literals to the `RegExp` type, similar to what is done for the numeric literals with `I32` etc. That would then print a different error, but already prepare for an eventual implementation.

**Pull request merged**